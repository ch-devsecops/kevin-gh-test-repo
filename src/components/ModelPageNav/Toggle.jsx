import React from 'react';
import { Icon, IconWrapper } from '@honda-canada/design-system-react';
import css from '@styled-system/css';
import styled from 'styled-components';

const ToggleButton = styled.button(
  css({
    border: 'none',
    background: 'transparent',
    backgroundImage: 'none',
    p: 'zero',
    m: 'zero',
    display: 'flex',
    alignItems: 'center',
  }),
);

const Arrow = styled(Icon)(({ rotated, theme }) =>
  css({
    transform: rotated ? 'rotate(180deg)' : 'rotate(0deg)',
    transition: `transform ${theme.transitionTimingFunction.default} ${theme.transitionDuration.t6}`,
  }),
);

export const Label = styled.span(({ color }) =>
  css({
    fontWeight: 'normal',
    fontFamily: 'bold',
    fontSize: '14px',
    color,
    letterSpacing: '0.5px',
  }),
);

const Toggle = ({ setIsOpen, label, isOpen, useToggle = true, backgroundColor }) => {
  if (useToggle) {
    return (
      <ToggleButton onClick={setIsOpen} color={backgroundColor === 'black' ? 'white' : 'black'}>
        {label && <Label color={backgroundColor === 'black' ? 'white' : 'black'}>{label}</Label>}
        <IconWrapper pl="xs" style={{ width: 'auto' }} color="white">
          <Arrow
            name="arrowDown"
            width="13px"
            height="6px"
            rotated={isOpen}
            color={backgroundColor === 'black' ? 'white' : 'black'}
          />
        </IconWrapper>
      </ToggleButton>
    );
  }
  return label ? <Label color={backgroundColor === 'black' ? 'white' : 'black'}>{label}</Label> : null;
};

export default Toggle;
