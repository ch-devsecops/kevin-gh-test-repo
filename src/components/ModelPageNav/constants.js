import { createQueryParam, getTransmissionKeyFromDefaultTransmission } from '../../utils/urls';

export const variant1 = 'variant1';
export const variant2 = 'variant2';

export const useConfiguration = variant => {
  let badgeSize;
  let zIndexWorkaroundClassName;
  let zIndexStringValue = 'modal';

  switch (variant) {
    case variant1:
      badgeSize = ['69px', '160px', '160px'];
      zIndexWorkaroundClassName = 'psp-z-index-workaround';
      zIndexStringValue = 'subNavigationBar';
      break;
    case variant2:
    default:
      badgeSize = '160px';
      zIndexWorkaroundClassName = 'acura-z-index-workaround honda-z-index-workaround';
      break;
  }
  return {
    badgeSize,
    zIndexWorkaroundClassName,
    zIndexStringValue,
  };
};

/**
 * Returns the URL for price and finance deeplinking for psp and build and price for auto.
 *
 * @variant {string} key - The app name
 * @baseUrl {string} The shared base url for auto in shared apps.
 * @params {string} The bapUrlparams
 * @urlTranslation {string} the price and finance url from sitecore dictionary
 * @vehicleType {string} the vehicle type on PSP
 * @category {string} the category of the selected trim
 * @model {string} the model of the selected trim
 */
export const getBapUrlParams = (variant, baseUrl, params, urlTranslation, vehicleType, category, selectedTrim) => {
  let bapUrlParams = `${baseUrl}${params}`;
  const urlParams = new URLSearchParams(bapUrlParams);
  const paramsObj = Object.fromEntries(urlParams);
  const categoryQueryParam = createQueryParam('category', category);
  const hideTrimAndModel = selectedTrim?.hidePriceAndFinance;

  const trimQueryParam = !hideTrimAndModel
    ? createQueryParam('trim', getTransmissionKeyFromDefaultTransmission(paramsObj?.transmission_key))
    : '';
  const yearQueryParam = createQueryParam('year', paramsObj?.model_year);
  const colorQueryParam = !hideTrimAndModel ? createQueryParam('color', paramsObj?.color_key) : '';
  const modelQueryParam = !hideTrimAndModel ? createQueryParam('model', selectedTrim?.detIdentifier) : '';

  switch (variant) {
    case variant1:
      bapUrlParams = `${urlTranslation}?vehicleType=${vehicleType}${categoryQueryParam}${yearQueryParam}${modelQueryParam}${trimQueryParam}${colorQueryParam}`;
      break;

    default:
      break;
  }
  return bapUrlParams;
};
