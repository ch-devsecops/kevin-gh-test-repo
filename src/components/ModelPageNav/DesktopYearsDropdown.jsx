import React, { useState, useEffect, useRef } from 'react';
import { Box, Link } from '@honda-canada/design-system-react';
import { NavLink } from 'react-router-dom';
import css from '@styled-system/css';
import styled from 'styled-components';
import Toggle, { Label } from './Toggle';
import { gtmSelectYear } from '../../utils/gtmEvents';

const linkHeight = 33;

const Years = styled(Box)(({ isOpen, theme, numYears }) =>
  css({
    height: isOpen ? `${numYears * linkHeight + 18}px` : 0,
    transition: `height ${theme.transitionTimingFunction.default} ${theme.transitionDuration.t6}`,
  }),
);

const YearLink = styled(Link)(({ selected, color, isOpen }) =>
  css({
    fontWeight: 'normal',
    fontFamily: 'bold',
    display: 'block',
    fontSize: '14px',
    lineHeight: '18px',
    textDecoration: 'none',
    color: selected ? 'primary' : color,
    visibility: isOpen ? 'vibible' : 'hidden',
    '&:hover': {
      color: 'primary',
    },
    letterSpacing: '0.5px',
  }),
);

const YearsDropdown = ({ years, selected, backgroundColor, isDark }) => {
  const [isOpen, setIsOpen] = useState(false);
  const selectedName = selected?.name;
  const dropdownRef = useRef();
  const linkColor = isDark ? 'white' : 'black';

  useEffect(() => {
    if (typeof window === 'undefined') return undefined;

    const handleEvent = e => {
      const shouldHandle = dropdownRef?.current && !dropdownRef.current.contains(e.target);

      if (shouldHandle) setIsOpen(false);
    };

    document.addEventListener('click', handleEvent);

    return () => {
      document.removeEventListener('click', handleEvent);
    };
  }, []);

  useEffect(() => {
    setIsOpen(false);
  }, [selectedName]);

  return (
    <Box position="relative" pl="m" ref={dropdownRef} data-testid="cy-model-nav-years-dropdown">
      <Toggle
        setIsOpen={() => {
          setIsOpen(!isOpen);
        }}
        isOpen={isOpen}
        label={selectedName && <Label color={linkColor}>{selectedName}</Label>}
        aria-haspopup="true"
        aria-expanded={isOpen}
        useToggle={years?.length > 1}
        backgroundColor={backgroundColor}
      />
      <Years
        as="ul"
        role="menu"
        isOpen={isOpen}
        numYears={years?.length}
        boxShadow="box.subNav"
        backgroundColor={isDark ? backgroundColor : 'grey.5'}
        overflow="hidden"
        position="absolute"
        top="61px"
        width="70px"
        left={0}
        border={isOpen ? '1px solid' : 'none'}
        borderTop="none"
        borderColor="grey.1"
      >
        {years?.map(year => (
          <Box
            as="li"
            pt="16px"
            key={year.url}
            px="m"
            display="flex"
            alignItems="center"
            role="menuitem"
            aria-label={year.name}
          >
            <YearLink
              data-testid="cy-model-nav-year-link"
              color={linkColor}
              {...(year.legacyPageUrl
                ? { as: 'a', href: year.legacyPageUrl }
                : { as: NavLink, to: year.correspondingUrl || year.url })}
              disableHover
              onClick={() => {
                gtmSelectYear(year?.name);
                setIsOpen(false);
              }}
              selected={selectedName && year.name === selectedName}
              isOpen={isOpen}
            >
              {year.name}
            </YearLink>
          </Box>
        ))}
      </Years>
    </Box>
  );
};

export default YearsDropdown;
