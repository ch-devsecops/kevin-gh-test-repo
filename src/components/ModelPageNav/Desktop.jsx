import React from 'react';
import { Box, Wrapper, Row, Column, Button, Link, Copy } from '@honda-canada/design-system-react';
import { NavLink } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import css from '@styled-system/css';
import styled from 'styled-components';
import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import YearsDropdown from './DesktopYearsDropdown';
import { HONDA_SITE_NAME } from '../../utils/constants';
import { getBapUrlParams } from './constants';
import { useOffersQueryParam } from './utils';

const ModelBadge = styled.img(
  css({
    ml: 'default',
  }),
);

const TrimBadge = styled.img`
  max-height: 11px;
  position: relative;
  top: -2px;
`;

const NavItem = styled.li(
  css({
    height: '100%',
    display: 'flex',
    alignItems: 'center',
  }),
);

const textStyles = {
  letterSpacing: '0.5px',
  fontWeight: 'regular',
  fontFamily: 'bold',
  fontSize: '14px',
  textDecoration: 'none',
};

const ModelBrandNameStyles = () =>
  css({
    ...textStyles,
    ml: 'l',
    lineHeight: ['18px', '22px'],
  });

const LinkStyles = ({ theme, color }) =>
  css({
    ...textStyles,
    color,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%',
    borderBottom: '3px solid transparent',
    boxSizing: 'border-box',
    '&:hover, &:focus': {
      color: 'primary',
      borderBottom: '3px solid transparent',
    },
    '&:focus-visible': {
      outline: `solid ${theme.colors.aodaFocused} 6px`,
    },
    // for activeClassName
    '&.active': {
      borderBottom: `3px solid ${theme.colors.primary}`,
    },
  });

const StyledExternalLink = styled(Link)(LinkStyles);

const StyledLink = styled(NavLink)(LinkStyles);

export const ModelBrandName = styled(Copy)(ModelBrandNameStyles);

const Desktop = ({
  specialTrimPage,
  years,
  selectedYear,
  modelBadge,
  navItems,
  bapBaseUrl,
  bapLabel,
  bapUrlParams,
  offersBaseUrl,
  offersUrlParams,
  modelName,
  bodyStyle,
  isDark,
  buildButtonStyling,
  sitecoreContext,
  showCTA,
  variant,
  vehicleType,
  selectedTrim,
  rendering,
}) => {
  const { t } = useTranslation();
  const { queryParam } = useOffersQueryParam(offersUrlParams);
  const gtmOverviewLabel = t('Pages.Models.Exploration.overviewNavLabel', { lng: 'en' })?.toLowerCase();
  const gtmOffersLabel = t('Pages.Models.Exploration.offersNavLabel', { lng: 'en' })?.toLowerCase();
  const { site } = sitecoreContext;
  const appName = site?.name;
  const backgroundColor = isDark ? 'black' : 'white';
  const textColor = isDark ? 'white' : 'black';

  return (
    <div>
      <Box
        display="flex"
        height={appName === HONDA_SITE_NAME ? '84px' : '75px'}
        alignItems="center"
        boxShadow={isDark ? 'darkBox.subNav' : 'box.subNav'}
        position="sticky"
        top={0}
        backgroundColor={backgroundColor}
        data-testid="cy-model-page-nav-desktop-wrapper"
      >
        <Wrapper gutters={[false, false]} height="100%">
          <Row height="100%">
            <Column width={(specialTrimPage ? 3 : 4) / 12} display="flex" alignItems="center" height="100%">
              <YearsDropdown years={years} selected={selectedYear} backgroundColor={backgroundColor} isDark={isDark} />
              {modelBadge?.src ? (
                <ModelBadge data-testid="cy-model-nav-name-badge" src={modelBadge.src} alt={modelBadge.alt} />
              ) : (
                <ModelBrandName data-testid="cy-model-nav-name-text-badge">{modelName}</ModelBrandName>
              )}
            </Column>
            <Column
              as="ul"
              width={(specialTrimPage ? 9 : 8) / 12}
              display="flex"
              alignItems="center"
              justifyContent="space-around"
            >
              {selectedYear && (
                <NavItem>
                  <StyledLink
                    color={textColor}
                    to={selectedYear.url}
                    activeClassName="active"
                    data-gtm-model={modelName}
                    data-gtm-body-style={bodyStyle}
                    data-gtm-interaction-type="primary model nav"
                    data-gtm-title={gtmOverviewLabel}
                    exact
                  >
                    {t('Pages.Models.Exploration.overviewNavLabel')}
                  </StyledLink>
                </NavItem>
              )}
              {navItems?.map(navItem => (
                <NavItem key={navItem.url}>
                  <StyledLink
                    data-testid="cy-model-nav-item"
                    color={textColor}
                    to={navItem.url}
                    activeClassName="active"
                    data-gtm-model={modelName}
                    data-gtm-body-style={bodyStyle}
                    data-gtm-interaction-type="primary model nav"
                    data-gtm-title={navItem.name?.toLowerCase()}
                    aria-label={
                      navItem.nameBadge?.item?.value?.src
                        ? navItem.nameBadge?.item?.value
                        : navItem.sectionNavLabel?.value || navItem.name
                    }
                  >
                    {navItem.nameBadge?.item?.value?.src ? (
                      <TrimBadge {...navItem.nameBadge?.item?.value} />
                    ) : (
                      navItem.sectionNavLabel?.value
                    )}
                  </StyledLink>
                </NavItem>
              ))}
              <NavItem>
                <StyledExternalLink
                  data-testid="cy-model-nav-offers"
                  color={textColor}
                  as="a"
                  href={`${offersBaseUrl}${queryParam}`}
                  data-gtm-model={modelName}
                  data-gtm-body-style={bodyStyle}
                  data-gtm-interaction-type="primary model nav"
                  data-gtm-title={gtmOffersLabel}
                  isLinkType
                >
                  {t('Pages.Models.Exploration.offersNavLabel')}
                </StyledExternalLink>
              </NavItem>
              <NavItem data-testid="cy-model-nav-price-and-finance">
                {showCTA ? (
                  <Button
                    as="a"
                    href={getBapUrlParams(
                      variant,
                      bapBaseUrl,
                      bapUrlParams,
                      t('Shared.Common.priceAndFinanceUrl'),
                      vehicleType,
                      modelName,
                      selectedTrim,
                    )}
                    styling={buildButtonStyling}
                    data-gtm-model={modelName}
                    data-gtm-body-style={bodyStyle}
                    data-gtm-interaction-type="cta: build"
                    data-gtm-component-type={rendering?.componentName}
                    data-gtm-title="build & price"
                    aria-label={bapLabel}
                  >
                    {bapLabel}
                  </Button>
                ) : null}
              </NavItem>
            </Column>
          </Row>
        </Wrapper>
      </Box>
    </div>
  );
};

export default withSitecoreContext()(Desktop);
