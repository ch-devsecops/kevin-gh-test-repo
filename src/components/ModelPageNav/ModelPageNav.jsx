import React, { useContext, useEffect } from 'react';
import { Media, Box } from '@honda-canada/design-system-react';
import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { useLocation } from 'react-router-dom';
import isValidUrlOrRelativeUrl from '@honda-canada/js-utilities/lib/isValidUrlOrRelativeUrl';
import isEqual from 'lodash/isEqual';
import isSSR from '../../utils/isSSR';
import { ModelExplorationContext } from '../ModelExplorationContext';
import Mobile from './Mobile';
import Desktop from './Desktop';
import { BUILD_AND_PRICE_URL, OFFERS_URL } from '../../utils/constants';
import useSharedApps from '../../utils/sitecoreContext/useSharedApps';
import { getModelNavPageBodyStyle, useZindexWorkaround } from './utils';
import { useConfiguration } from './constants';
import BottomElementContext from '../Footer/BottomElementProvider/BottomElementContext';

const getBuildButtonStyling = (isDark, isTrimSpecial) => (isDark || isTrimSpecial ? 'special' : 'primary');

/**
 * Returns a url that corresponds to the currently-displayed page. For example,
 * when the current path is /ILX/Trims, the 2021 dropdown item should link to
 * /ILX/2021/Trims instead of just /ILX/2021.
 * @param {string} year - a model year
 * @param {array} navItems - an array of nav items, each with a url
 * @param {string} pathname - the current pathname, from react-router-dom
 * @returns a corresponding URL for the given year
 */
const getCorrespondingUrl = (year, navItems, pathname = '') => {
  const currentPage = navItems?.find(navItem => pathname?.toLowerCase()?.includes(navItem?.url?.toLowerCase()));
  const currentPageName = currentPage?.name;
  const correspondingSubPage = year.subPages?.find(s => s.name === currentPageName);

  return correspondingSubPage?.url;
};

const getLegacyPageUrl = (year, years) => {
  const foundYear = years?.find(v => v.name === year.name);
  return foundYear?.legacyPageUrl?.value || undefined;
};

const getNodes = items => {
  if (!items) return {};
  return items.map(item => ({
    name: item.name,
    url: item.url,
  }));
};

const ModelPageNav = ({ fields, sitecoreContext, rendering, categoryName, modelName, variant }) => {
  const { setIsHidden, setHeaderHiddenOnScroll } = useContext(BottomElementContext);
  const modelExplorationContext = useContext(ModelExplorationContext) || {};

  const { pathname } = useLocation();

  const bapPathUrl = useSharedApps(BUILD_AND_PRICE_URL);

  const offersBaseUrl = useSharedApps(OFFERS_URL);
  const vehicleType = fields?.data?.value?.modelPageSettings?.modelCategory?.fields?.vehicleType?.name;

  const zIndexWorkaround = useZindexWorkaround();

  const { zIndexWorkaroundClassName, zIndexStringValue } = useConfiguration(variant);

  useEffect(() => {
    if (isSSR()) return undefined;
    if (typeof setHeaderHiddenOnScroll === 'function') setHeaderHiddenOnScroll(true);

    return () => {
      // This function will be called when the component is unmounted
      setIsHidden(false);
      if (typeof setHeaderHiddenOnScroll === 'function') setHeaderHiddenOnScroll(false);
    };
  }, []);

  if (!fields) return null;

  const { modelNav, modelPageSettings } = fields?.data?.value || {};
  const { pageEditing } = sitecoreContext;

  const isDark = modelPageSettings?.isDarkMode?.value === '1';

  const navItems = modelNav?.map(navItem => {
    // If a nav item has no nameBadge image, ExperienceEditor adds a placeholder image,
    // which we don't want because it prevents the text value from showing up.
    if (navItem.nameBadge?.item?.editable && navItem.nameBadge?.item?.value?.class === 'scEmptyImage') {
      // eslint-disable-next-line no-param-reassign
      delete navItem.nameBadge.item.value.src;
    }

    return navItem;
  });

  const { navItems: navigationItems, setNavItems } = modelExplorationContext;
  if (!isEqual(getNodes(navItems), getNodes(navigationItems))) {
    setNavItems(getNodes(navItems));
  }

  const years = fields.data?.value?.years?.items?.map(year => ({
    ...year,
    legacyPageUrl: getLegacyPageUrl(year, categoryName?.fields?.modelYears),
    correspondingUrl: getCorrespondingUrl(year, navItems, pathname),
  }));

  const modelBadge = modelPageSettings?.nameBadge?.item?.value;
  const specialTrimPage = navItems?.find(navItem => navItem.nameBadge?.item?.value?.src);
  const ctaLink = isValidUrlOrRelativeUrl(modelPageSettings?.ctaLink?.item?.value?.href)
    ? modelPageSettings?.ctaLink?.item?.value
    : undefined;
  const showCTA = !!ctaLink?.href;
  const isNsx = modelPageSettings?.isNsx?.value;
  const bapBaseUrl = isNsx ? ctaLink?.href : bapPathUrl;
  const bapLabel = ctaLink?.text;

  const props = {
    showCTA,
    modelName,
    specialTrimPage,
    years,
    modelBadge,
    navItems,
    bapBaseUrl,
    bapLabel,
    offersBaseUrl,
  };

  return (
    <ModelExplorationContext.Consumer>
      {({ bapUrlParams, setIsModelNavOpen, modelYear, offersUrlParams, selectedTrim }) => {
        // get bodyStyle from defaultTrim for modelYear
        // here
        const defaultModelYear = categoryName?.fields?.modelYears?.find(year => year.name === modelYear);
        const { bodyStyle } = getModelNavPageBodyStyle(variant, defaultModelYear, vehicleType);
        return (
          <>
            {zIndexWorkaround}
            <Box
              data-gtm-category="model interactions"
              data-gtm-component-type={rendering?.componentName}
              className={zIndexWorkaroundClassName}
              zIndex={pageEditing ? 210 : zIndexStringValue}
              backgroundColor={isDark ? 'black' : 'white'}
              position="sticky"
              top={0}
            >
              <Media lessThan="desktop">
                <Mobile
                  {...props}
                  bapUrlParams={isNsx ? '' : bapUrlParams}
                  offersUrlParams={offersUrlParams}
                  setIsModelNavOpen={setIsModelNavOpen}
                  selectedYear={years?.find(y => y.name === modelYear)}
                  isDark={isDark}
                  bodyStyle={bodyStyle}
                  buildButtonStyling={getBuildButtonStyling(isDark, selectedTrim?.isSpecialType)}
                  variant={variant}
                  vehicleType={vehicleType}
                  selectedTrim={selectedTrim}
                  rendering={rendering}
                />
              </Media>
              <Media greaterThanOrEqual="desktop">
                <Desktop
                  {...props}
                  bapUrlParams={isNsx ? '' : bapUrlParams}
                  isDark={isDark}
                  offersUrlParams={offersUrlParams}
                  selectedYear={years?.find(y => y.name === modelYear)}
                  bodyStyle={bodyStyle}
                  buildButtonStyling={getBuildButtonStyling(isDark, selectedTrim?.isSpecialType)}
                  variant={variant}
                  vehicleType={vehicleType}
                  selectedTrim={selectedTrim}
                  rendering={rendering}
                />
              </Media>
            </Box>
          </>
        );
      }}
    </ModelExplorationContext.Consumer>
  );
};

export default withSitecoreContext()(ModelPageNav);
