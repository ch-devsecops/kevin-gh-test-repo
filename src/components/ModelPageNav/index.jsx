import React from 'react';
import ModelPageNav from './ModelPageNav';
import { PSP_SITE_NAME, MC_SITE_NAME } from '../../utils/constants';
import { useAppName } from '../../utils/sitecoreContext';
import { variant1, variant2 } from './constants';

const ModelPageNavWrapper = ({ fields, rendering }) => {
  const appName = useAppName();

  let categoryName;
  let modelName;
  let variant;

  switch (appName) {
    case PSP_SITE_NAME:
    case MC_SITE_NAME:
      categoryName = fields?.data?.value?.modelPageSettings?.modelCategory;
      modelName = categoryName?.fields?.categoryName?.value;
      variant = variant1;
      break;
    default:
      categoryName = fields?.data?.value?.modelPageSettings?.model;
      modelName = categoryName?.fields?.name?.toLowerCase();
      variant = variant2;
  }

  return (
    <ModelPageNav
      variant={variant}
      fields={fields}
      rendering={rendering}
      categoryName={categoryName}
      modelName={modelName}
    />
  );
};

export default ModelPageNavWrapper;
