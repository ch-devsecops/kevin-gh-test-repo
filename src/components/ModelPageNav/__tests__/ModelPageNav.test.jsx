import React from 'react';
import { jssRender, cleanup } from 'test-utils';
import ModelPageNav from '../index';

const mockData = require('./mockData.test.json');

afterEach(cleanup);

describe('<ModelPageNav>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<ModelPageNav fields={mockData.fields} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should render as null without fields', () => {
    const { container } = jssRender(<ModelPageNav />);

    expect(container.firstChild).toBeNull();
  });
});
