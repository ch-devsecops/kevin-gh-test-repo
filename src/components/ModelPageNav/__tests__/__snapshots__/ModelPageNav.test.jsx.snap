// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelPageNav> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 iCZnBB acura-z-index-workaround honda-z-index-workaround"
    data-gtm-category="model interactions"
  >
    <div
      class="fresnel-container fresnel-lessThan-desktop "
    >
      <div
        class="Box-sc-1bsf784-0 bMfbYL"
        display="flex"
      >
        <div
          class="Box-sc-1bsf784-0 eTnRTd"
          data-testid="cy-model-nav-years-dropdown"
          display="flex"
          height="70px"
        >
          <button
            class="sc-bdVaJa fDjyJc"
            color="black"
          >
            <div
              class="IconWrapper-lp9q2r-0 kUnDYo"
              color="white"
              style="width: auto;"
            />
          </button>
          <img
            alt="ilx"
            class="sc-gZMcBi hQmOSK"
            data-testid="cy-model-nav-name-badge"
            src="https://app-acura-pod02-cd-plan-dmm-ccentral-uat.azurewebsites.net/-/media/Acura/ILX/badge.png?h=28&w=190&hash=7284B16F08D71C588F2E3D87F96B161C"
          />
        </div>
        <div
          class="Box-sc-1bsf784-0 eTnRTd"
          data-testid="cy-model-nav-navigation-dropdown"
          display="flex"
          height="70px"
        >
          <button
            class="sc-bdVaJa fDjyJc"
            color="black"
          >
            <span
              class="sc-htpNat bwTdfq"
              color="black"
            >
              <span
                class="sc-htpNat bwTdfq"
                color="black"
              >
                Pages.Models.Exploration.overviewNavLabel
              </span>
            </span>
            <div
              class="IconWrapper-lp9q2r-0 kUnDYo"
              color="white"
              style="width: auto;"
            />
          </button>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 sc-jTzLTM hPXCIA"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 feLJFE sc-fjdhpX dmtGjr"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hsvmmC"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 lbWKvq"
            >
              <p
                class="sc-1k44dfv-0 bZVSot"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                Pages.Models.Exploration.mobileLandscapeToastLabel
              </p>
            </div>
            <button
              class="sc-1k44dfv-0 rejBv"
            >
              <div
                class="IconWrapper-lp9q2r-0 bbbwgU"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 dTZDkL Box-sc-1bsf784-0 sc-1k44dfv-0 bwptLY"
        direction="down"
      >
        <ul
          data-testid="cy-model-nav-mobile-years-list"
        >
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="sc-gqjmRU kNrAez"
              color="black"
              data-testid="cy-model-nav-year-link"
              href="/en/ILX/2021"
              last="false"
            >
              2021
            </a>
          </li>
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="sc-gqjmRU fZMWgg"
              color="black"
              data-testid="cy-model-nav-year-link"
              href="/en/ILX/2022"
              last="true"
            >
              2022
            </a>
          </li>
        </ul>
      </div>
      <div
        class="Box-sc-1bsf784-0 NjloG Box-sc-1bsf784-0 sc-1k44dfv-0 bIaHCG"
        direction="down"
      >
        <ul
          data-testid="cy-model-nav-mobile-nav-item-list"
        >
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          />
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="sc-gqjmRU kNrAez"
              color="black"
              data-gtm-body-style=""
              data-gtm-interaction-type="primary model nav"
              data-testid="cy-model-nav-item"
              href="/en/ILX/2022/Trims"
            >
              Trims
            </a>
          </li>
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="sc-gqjmRU kNrAez"
              color="black"
              data-gtm-body-style=""
              data-gtm-interaction-type="primary model nav"
              data-testid="cy-model-nav-item"
              href="/en/ILX/2022/Specifications"
            >
              Specifications
            </a>
          </li>
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="sc-gqjmRU kNrAez"
              color="black"
              data-gtm-body-style=""
              data-gtm-interaction-type="primary model nav"
              data-testid="cy-model-nav-item"
              href="/en/ILX/2022/Accessories"
            >
              Accessories
            </a>
          </li>
          <li
            class="Box-sc-1bsf784-0 VbvFZ"
            display="flex"
          >
            <a
              class="Link-sc-16wd4a0-0 sc-VigVT kDERDW"
              color="black"
              data-gtm-body-style=""
              data-gtm-interaction-type="primary model nav"
              data-gtm-title="pages.models.exploration.offersnavlabel"
              href="undefined"
            >
              Pages.Models.Exploration.offersNavLabel
            </a>
          </li>
          <li
            class="Box-sc-1bsf784-0 gSaKDL"
            display="flex"
          >
            <a
              aria-label="Build & Price"
              class="Button-pulw6t-0 fePOwb"
              data-gtm-body-style=""
              data-gtm-interaction-type="cta: build"
              data-gtm-title="build & price"
              data-testid="dsr-button-component"
              href="undefined"
            >
              Build & Price
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-desktop "
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 eDJlFb"
          data-testid="cy-model-page-nav-desktop-wrapper"
          display="flex"
          height="75px"
        >
          <div
            class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 jyBXfB"
            height="100%"
          >
            <div
              class="Box-sc-1bsf784-0 ekugtK"
              display="flex"
              height="100%"
              width="100%"
            >
              <div
                class="Box-sc-1bsf784-0 Column-t2p57o-0 ibvhkG"
                display="flex"
                height="100%"
                width="0.3333333333333333"
              >
                <div
                  class="Box-sc-1bsf784-0 kfGPDj"
                  data-testid="cy-model-nav-years-dropdown"
                >
                  <button
                    class="sc-bdVaJa fDjyJc"
                    color="black"
                  >
                    <div
                      class="IconWrapper-lp9q2r-0 kUnDYo"
                      color="white"
                      style="width: auto;"
                    />
                  </button>
                  <ul
                    class="Box-sc-1bsf784-0 sc-bxivhb gsRSFQ"
                    overflow="hidden"
                    role="menu"
                    width="70px"
                  >
                    <li
                      aria-label="2021"
                      class="Box-sc-1bsf784-0 eclhwq"
                      display="flex"
                      role="menuitem"
                    >
                      <a
                        class="Link-sc-16wd4a0-0 sc-ifAKCX blWNRV"
                        color="black"
                        data-testid="cy-model-nav-year-link"
                        href="/en/ILX/2021"
                      >
                        2021
                      </a>
                    </li>
                    <li
                      aria-label="2022"
                      class="Box-sc-1bsf784-0 eclhwq"
                      display="flex"
                      role="menuitem"
                    >
                      <a
                        class="Link-sc-16wd4a0-0 sc-ifAKCX blWNRV"
                        color="black"
                        data-testid="cy-model-nav-year-link"
                        href="/en/ILX/2022"
                      >
                        2022
                      </a>
                    </li>
                  </ul>
                </div>
                <img
                  alt="ilx"
                  class="sc-EHOje evdWlB"
                  data-testid="cy-model-nav-name-badge"
                  src="https://app-acura-pod02-cd-plan-dmm-ccentral-uat.azurewebsites.net/-/media/Acura/ILX/badge.png?h=28&w=190&hash=7284B16F08D71C588F2E3D87F96B161C"
                />
              </div>
              <ul
                class="Box-sc-1bsf784-0 Column-t2p57o-0 eNEuoU"
                display="flex"
                width="0.6666666666666666"
              >
                <li
                  class="sc-gzVnrw iheFIc"
                >
                  <a
                    aria-label="Trims"
                    class="sc-dnqmqq fIOQzR"
                    color="black"
                    data-gtm-body-style=""
                    data-gtm-interaction-type="primary model nav"
                    data-testid="cy-model-nav-item"
                    href="/en/ILX/2022/Trims"
                  >
                    Trims
                  </a>
                </li>
                <li
                  class="sc-gzVnrw iheFIc"
                >
                  <a
                    aria-label="Specifications"
                    class="sc-dnqmqq fIOQzR"
                    color="black"
                    data-gtm-body-style=""
                    data-gtm-interaction-type="primary model nav"
                    data-testid="cy-model-nav-item"
                    href="/en/ILX/2022/Specifications"
                  >
                    Specifications
                  </a>
                </li>
                <li
                  class="sc-gzVnrw iheFIc"
                >
                  <a
                    aria-label="Accessories"
                    class="sc-dnqmqq fIOQzR"
                    color="black"
                    data-gtm-body-style=""
                    data-gtm-interaction-type="primary model nav"
                    data-testid="cy-model-nav-item"
                    href="/en/ILX/2022/Accessories"
                  >
                    Accessories
                  </a>
                </li>
                <li
                  class="sc-gzVnrw iheFIc"
                >
                  <a
                    class="Link-sc-16wd4a0-0 sc-htoDjs djzwKt"
                    color="black"
                    data-gtm-body-style=""
                    data-gtm-interaction-type="primary model nav"
                    data-gtm-title="pages.models.exploration.offersnavlabel"
                    data-testid="cy-model-nav-offers"
                    href="undefined"
                  >
                    Pages.Models.Exploration.offersNavLabel
                  </a>
                </li>
                <li
                  class="sc-gzVnrw iheFIc"
                  data-testid="cy-model-nav-price-and-finance"
                >
                  <a
                    aria-label="Build & Price"
                    class="Button-pulw6t-0 fePOwb"
                    data-gtm-body-style=""
                    data-gtm-interaction-type="cta: build"
                    data-gtm-title="build & price"
                    data-testid="dsr-button-component"
                    href="undefined"
                  >
                    Build & Price
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
